paimon-web-api/src/main/java/org/apache/paimon/web/api/table/TableChange.java

```diff
@@ -0,0 +1,486 @@
+        return new DropColumn(columnName);
+    }
+
+    static SetOption set(String key, String value) {
+        return new SetOption(key, value);
+    }
+
+    static ResetOption reset(String key) {
+        return new ResetOption(key);
+    }
+
+    /** A table change to add a column. */
+    class AddColumn implements TableChange {
+
+        private final ColumnMetadata column;
+        private final ColumnPosition position;
+
+        private AddColumn(ColumnMetadata column, ColumnPosition position) {
+            this.column = column;
+            this.position = position;
+        }
+
+        public ColumnMetadata getColumn() {
+            return column;
+        }
+
+        @Nullable
+        public ColumnPosition getPosition() {
+            return position;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof AddColumn)) {
+                return false;
+            }
+            AddColumn addColumn = (AddColumn) o;
+            return Objects.equals(column, addColumn.column)
+                    && Objects.equals(position, addColumn.position);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(column, position);
+        }
+
+        @Override
+        public String toString() {
+            return "AddColumn{" + "column=" + column + ", position=" + position + '}';
+        }
+    }
+
+    /** A base schema change to modify a column. */
+    class ModifyColumn implements TableChange {
+
+        protected final ColumnMetadata oldColumn;
+        protected final ColumnMetadata newColumn;
+
+        protected final @Nullable ColumnPosition newPosition;
+
+        public ModifyColumn(
+                ColumnMetadata oldColumn,
+                ColumnMetadata newColumn,
+                @Nullable ColumnPosition newPosition) {
+            this.oldColumn = oldColumn;
+            this.newColumn = newColumn;
+            this.newPosition = newPosition;
+        }
+
+        /** Returns the original {@link ColumnMetadata} instance. */
+        public ColumnMetadata getOldColumn() {
+            return oldColumn;
+        }
+
+        /** Returns the modified {@link ColumnMetadata} instance. */
+        public ColumnMetadata getNewColumn() {
+            return newColumn;
+        }
+
+        /**
+         * Returns the position of the modified {@link ColumnMetadata} instance. When the return
+         * value is null, it means modify the column at the original position. When the return value
+         * is FIRST, it means move the modified column to the first. When the return value is AFTER,
+         * it means move the column after the referred column.
+         */
+        public @Nullable ColumnPosition getNewPosition() {
+            return newPosition;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
```

Comment:
What is the return value of `equals` when `o` is null?

Reply:
I think when o is null, !(o instanceof xxx) this condition will return true and then the whole equals method will return false.